File: GUISM_SERVER/source/habito/Habito.java
package habito;

import especializacao.Especializacao;
import kikaha.jdbi.serializers.Column;
import kikaha.jdbi.serializers.Entity;
import kikaha.jdbi.serializers.Optional;
import lombok.Data;

import java.util.Set;

@Data
@Entity
public class Habito {

    @Column
    Long idHabito;

    @Column
    String nomeHabito;

    @Column
    String descHabito;

    @Column
    @Optional
    Long qtdFichaHabito;

    Set<Especializacao> especializacoes;
}

File: src/components/habito/Habito.ts
import Especializacao from "../especializacao/Especializacao";


export default interface Habito {
    idHabito: number;
    nomeHabito?: string;
    descHabito?: string;
    qtdFichaHabito?: number;
    especializacoes?: Especializacao[];
}
